package/harvester-os/files/usr/sbin/gen-kdump-initrd

```diff
@@ -0,0 +1,52 @@
+#!/bin/bash -ex
+
+KERNEL_VERSION=$(uname -r)
+INITRD_KDUMP_NAME="initrd-$KERNEL_VERSION-kdump"
+INITRD_CACHE_DIR="/usr/local/cache/kdump"
+
+if [ -e /boot/$INITRD_KDUMP_NAME ]; then
+  echo /boot/$INITRD_KDUMP_NAME exists. Skip generating a new one.
+  exit 0
+fi
+
+if [ -e $INITRD_CACHE_DIR/$INITRD_KDUMP_NAME ]; then
+  echo "Using cache: $INITRD_CACHE_DIR/$INITRD_KDUMP_NAME"
+  cp $INITRD_CACHE_DIR/$INITRD_KDUMP_NAME /boot/$INITRD_KDUMP_NAME
+  exit 0
+fi
+
+mkdir -p $INITRD_CACHE_DIR
+
+TMP_DIR=$(mktemp -d)
+trap "rm -rf $TMP_DIR" exit
+
+TMP_INITRD=$TMP_DIR/initrd
+INITRD_EXTRACTED_DIR=$TMP_DIR/extracted
+DRACUT_CONFDIR=$TMP_DIR/dracut.conf.d
+
+mkdir -p $INITRD_EXTRACTED_DIR
+mkdir -p $DRACUT_CONFDIR
+
+
+# specify a new dracut config dir to avoid including cOS configs in /etc/dracut.conf.d
+cat > $DRACUT_CONFDIR/99-tmp.conf <<EOF
+early_microcode="no"
+host_only="yes"
+EOF
+/usr/bin/dracut --confdir $DRACUT_CONFDIR --force --omit "plymouth resume usrmount" --add kdump $TMP_INITRD $KERNEL_VERSION
```

Comment:
Not a big deal. Just curious about little stuff:

- How were the omitted modules be determine?
- Could we pass `--hostonly` and `--no-early-microcode` instead of creating a conf file?

Reply:
They are the same thing, but since I need to override the confdir, I put those in the config files.
It also shortens the command and makes adding new params easier in the future.